server/main.go

```diff
@@ -0,0 +1,198 @@
+	Success        bool                   `json:"success"`
+	IgnitionConfig map[string]interface{} `json:"ignition_config"`
+	ErrLines       []int                  `json:"err_lines"`
+	ErrMsg         string                 `json:"err_msg"`
+}
+
+// write the response and sets the status
+func writeResponse(w *http.ResponseWriter, res responseData, status int) {
+	resJSON, err := json.Marshal(res)
+	if err != nil {
+		http.Error(*w, "Failed to parse struct `responseData` into JSON object", http.StatusInternalServerError)
+	}
+
+	(*w).Header().Set("Content-Type", "application/json")
+	(*w).WriteHeader(status)
+	(*w).Write(resJSON)
+}
+
+// runFCCT runs the FCCT command and returns the error, stdout, and stderr
+func runFCCT(input string) (string, string, error) {
+	var stdout bytes.Buffer
+	var stderr bytes.Buffer
+	cmd := exec.Command(fcct)
+	cmd.Stdin = strings.NewReader(input)
+	cmd.Stdout = &stdout
+	cmd.Stderr = &stderr
+	err := cmd.Run()
+	return stdout.String(), stderr.String(), err
+}
+
+// logMiddleware produces simple logs when wrapped around a HandlerFunc
+func logMiddleware(handler http.HandlerFunc) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		log.Printf("%s - %s - %s", r.RemoteAddr, r.Method, r.URL.Path)
+		if debug {
+			log.Printf("Request: %+v", r)
+		}
+		// Execute the original handler
+		handler(w, r)
+	}
+}
+
+// configHandler handles config requests when they come as a POST
+func configHandler(w http.ResponseWriter, r *http.Request) {
+	// enable CORS
+	w.Header().Set("Access-Control-Allow-Origin", "*")
+	w.Header().Set("Access-Control-Allow-Methods", "POST, OPTIONS")
+	w.Header().Set("Access-Control-Allow-Headers", "Accept, Content-Type, Content-Length, Accept-Encoding, X-CSRF-Token, Authorization")
+
+	// note: to enable CORS, it should handle OPTIONS as well
+	if r.Method == "OPTIONS" {
+		return
+	}
+
+	// If it's a POST, process
+	if r.Method == "POST" {
+		// https://golang.org/pkg/encoding/json/
+		// https://golang.org/pkg/os/exec/ (or import the golang code from fcct)
+
+		pd := postData{ConfigString: "default"}
+		res := responseData{false, nil, nil, ""}
+
+		contentType := r.Header.Get("Content-type")
+		if !strings.Contains(contentType, "application/json") {
+			res.ErrMsg = "Content-type must contain 'application/json'"
+			log.Printf(res.ErrMsg)
+			writeResponse(&w, res, http.StatusBadRequest)
+			return
+		}
+
+		decoder := json.NewDecoder(r.Body)
+		decoder.DisallowUnknownFields()
+		err := decoder.Decode(&pd)
+		if err != nil {
+			log.Println(err)
+			res.ErrMsg = fmt.Sprintf("%s", err)
+			writeResponse(&w, res, http.StatusInternalServerError)
+			return
+		}
+
+		// checks if POST json contains a `config_string` key
+		if pd.ConfigString == "default" {
+			res.ErrMsg = "failed: config_string must be set"
+			log.Printf(res.ErrMsg)
+			writeResponse(&w, res, http.StatusBadRequest)
+			return
+		}
+
+		// checks if `config_string` has reasonal length
+		maxLen := 31415
```

Comment:
In the future (not now) this should probably be set at run time instead of reset on every call.